sdk/storage/inc/datalake/datalake_options.hpp

```diff
@@ -0,0 +1,1030 @@
+     *        the blob and returned with a read request.
+     */
+    std::string ContentLanguage;
+
+    /**
+     * @brief Specify an ETag value to operate only on path with a matching value.
+     */
+    std::string IfMatch;
+
+    /**
+     * @brief Specify an ETag value to operate only on path without a matching value.
+     */
+    std::string IfNoneMatch;
+
+    /**
+     * @brief A date and time value. Specify this header to perform the
+     *        operation only if the resource has been modified since the
+     (        specified date and time.
+    */
+    std::string IfModifiedSince;
+
+    /**
+     * @brief A date and time value. Specify this header to perform the
+     *        operation only if the resource has not been modified since
+     *        the specified date and time.
+     */
+    std::string IfUnmodifiedSince;
+
+    /**
+     * @brief User-defined metadata to be stored with the path. Note that the string may only
+     *        contain ASCII characters in the ISO-8859-1 character set. If the filesystem exists,
+     *        any metadata not included in the list will be removed. All metadata are removed if
+     *        the value is null. To merge new and existing metadata, first get all existing metadata
+     *        and the current E-Tag, then make a conditional request with the E-Tag and include
+     *        values for all metadata.
+     */
+    std::map<std::string, std::string> Metadata;
+
+    /**
+     * @brief The owner of the blob or directory.
+     */
+    std::string Owner;
+
+    /**
+     * @brief The owning group of the blob or directory.
+     */
+    std::string Group;
+
+    /**
+     * @brief only valid if Hierarchical Namespace is enabled for the account. Sets POSIX
+     *        access permissions for the file owner, the file owning group, and others.
+     *        Each class may be granted read, write, or execute permission.
+     *        The sticky bit is also supported.  Both symbolic (rwxrw-rw-) and 4-digit octal
+     *        notation (e.g. 0766) are supported.
+     */
+    std::string Permissions;
+
+    /**
+     * @brief Sets POSIX access control rights on files and directories. The value is a
+     *        comma-separated list of access control entries. Each access control entry (ACE)
+     *        consists of a scope, a type, a user or group identifier, and permissions in the
+     *        format "[scope:][type]:[id]:[permissions]".
+     */
+    std::string Acl;
+
+    /**
+     * @brief The data to be appended to the resource.
+     */
+    std::vector<uint8_t> Data;
+  };
+
+  /**
+   * @brief Optional parameters for DataLakePathClient::Create
+   * @remark Some optional parameter is mandatory in certain combination.
+   *         More details:
+   * https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/create
+   */
+  struct PathCreateOptions : public SharedRequestOptions
```

Comment:
Rename and Create should be two functions in convenience layer.

Reply:
Separated in the latest iteration.